commons/src/config/output.rs

```diff
@@ -0,0 +1,150 @@
+use std::{collections::HashSet, path::PathBuf, str::FromStr};
+
+use clap::{Args, ValueEnum};
+use logid::set_event_with;
+use serde::{Deserialize, Serialize};
+
+use crate::log_id::COMMONS_LOG_ID_MAP;
+
+use super::{log_id::ConfigErrLogId, parse_to_hashset, ConfigFns, ReplaceIfNone};
+
+#[derive(Args, Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Output {
+    #[arg(long = "output-file")]
+    pub file: Option<PathBuf>,
+    #[arg(long, alias = "output-formats", value_parser = parse_to_hashset::<OutputFormat>)]
+    pub formats: HashSet<OutputFormat>,
+    #[command(flatten)]
+    #[serde(flatten)]
+    pub format_specific: OutputFormatSpecific,
+    /// `true` overwrites existing output files
+    #[arg(long, alias = "overwrite-out-files")]
+    pub overwrite: bool,
+}
+
+impl ConfigFns for Output {
+    fn merge(&mut self, other: Self) {
+        self.file.replace_none(other.file);
+        self.formats.extend(other.formats.iter());
+        self.format_specific.merge(other.format_specific);
+    }
+
+    fn validate(&self) -> Result<(), logid::log_id::LogId> {
+        self.format_specific.validate()?;
+
+        if self.formats.is_empty() {
+            return Err(set_event_with!(
+                ConfigErrLogId::InvalidConfig,
+                &COMMONS_LOG_ID_MAP,
+                "No output format was set."
+            )
+            .into());
+        }
+
+        if let Some(ref file) = self.file {
+            let mut filepath = file.clone();
+            for format in &self.formats {
+                filepath.set_extension(format.extension());
+
+                if filepath.exists() && !self.overwrite {
```

Comment:
Small optimization possible (although maybe not necessary, since this part won't be run a lot, only at the start): we could pull the `!self.overwrite` check outside of the `for` loop. The whole loop can be skipped if `self.overwrite == true`.